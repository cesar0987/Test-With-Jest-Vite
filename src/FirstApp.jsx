import PropTypes from 'prop-types';
const FirstApp = ({title,subTitle,name}) => {
    return (
        <>
        <h1>{title}</h1>
        <p>{subTitle}</p>
        <p>{name}</p>
        </>
    );
    }

export default FirstApp;

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired
}