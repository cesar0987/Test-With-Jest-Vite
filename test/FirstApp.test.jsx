import { render } from '@testing-library/react';
import FirstApp from '../src/FirstApp';
describe('Prueba en <FirstApp/>', () => {

    test('debe hacer match con el snapshot', () => {

        const title = 'Hola soy Fernando';
        const subTitle = 'Soy un subtitulo';
        const name = 'Cesar Fernando';
        const { container } = render( <FirstApp title={title} subTitle={subTitle} name={name}/> );

        expect( container ).toMatchSnapshot();

 })

    test('debe de mostrar el titulo en un h1'), () => {
            
            const title = 'Hola soy Fernando';
            const subTitle = 'Soy un subtitulo';
            const name = 'Cesar Fernando';
            const { getByText } = render( <FirstApp title={title} subTitle={subTitle} name={name}/> );
    
            expect( getByText(title) ).toBeInTheDocument();
    
        }
})