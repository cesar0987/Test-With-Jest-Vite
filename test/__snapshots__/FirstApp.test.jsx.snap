// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <FirstApp/> debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola soy Fernando
  </h1>
  <p>
    Soy un subtitulo
  </p>
  <p>
    Cesar Fernando
  </p>
</div>
`;
